pkg/envoy/secrets/secrets.go

```diff
@@ -63,3 +67,25 @@ func (sdsc *SDSCert) GetMeshService() (*service.MeshService, error) {
 		Name:      slices[1],
 	}, nil
 }
+
+// GetK8sServiceAccount unmarshals a K8sServiceAccount type from a SDSCert name
+func (sdsc *SDSCert) GetK8sServiceAccount() (*identity.K8sServiceAccount, error) {
+	slices := strings.Split(sdsc.Name, namespaceNameSeparator)
+	if len(slices) != 2 {
+		log.Error().Msgf("Error converting Service Account %s from string to K8sServiceAccount", sdsc.Name)
+		return nil, identity.ErrInvalidNamespacedServiceStringFormat
+	}
+
+	// Make sure the slices are not empty. Split might actually leave empty slices.
+	for _, sep := range slices {
+		if len(sep) == 0 {
```

Comment:
Instead of a loop, why not:
```
if slices[0] == "" || slices[1] == ""
```
given that L74 guarantees 2 elements in the slice.

Reply:
Updated! Good idea